sharding/indexHashedNodesCoordinator.go

```diff
@@ -17,43 +17,42 @@ type indexHashedNodesCoordinator struct {
 	nodesMap                map[uint32][]Validator
 	shardConsensusGroupSize int
 	metaConsensusGroupSize  int
+	selfPubKey              []byte
 }
 
 // NewIndexHashedNodesCoordinator creates a new index hashed group selector
-func NewIndexHashedNodesCoordinator(
-	shardConsensusGroupSize int,
-	metaConsensusGroupSize int,
-	hasher hashing.Hasher,
-	shardId uint32,
-	nbShards uint32,
-	nodes map[uint32][]Validator,
-) (*indexHashedNodesCoordinator, error) {
-	if shardConsensusGroupSize < 1 || metaConsensusGroupSize < 1 {
+func NewIndexHashedNodesCoordinator(arguments ArgNodesCoordinator) (*indexHashedNodesCoordinator, error) {
+	if arguments.ShardConsensusGroupSize < 1 || arguments.MetaConsensusGroupSize < 1 {
 		return nil, ErrInvalidConsensusGroupSize
 	}
 
-	if nbShards < 1 {
+	if arguments.NbShards < 1 {
 		return nil, ErrInvalidNumberOfShards
 	}
 
-	if shardId >= nbShards && shardId != MetachainShardId {
+	if arguments.ShardId >= arguments.NbShards && arguments.ShardId != MetachainShardId {
 		return nil, ErrInvalidShardId
 	}
 
-	if hasher == nil {
+	if arguments.Hasher == nil {
```

Comment:
you can create a func where you check all parameters of the `arguments`

Reply:
Good suggestion. Done.